chain/chain/src/store.rs

```diff
@@ -2211,13 +2212,45 @@ impl<'a> ChainStoreUpdate<'a> {
         Ok(())
     }
 
+    pub fn gc_outgoing_receipts(&mut self, block_hash: &CryptoHash, shard_id: ShardId) {
+        let mut store_update = self.store().store_update();
+        match self.get_outgoing_receipts(block_hash, shard_id).map(|receipts| {
+            receipts.iter().map(|receipt| receipt.receipt_id.clone()).collect::<Vec<_>>()
+        }) {
+            Ok(receipt_ids) => {
+                for receipt_id in receipt_ids {
+                    let key: Vec<u8> = receipt_id.into();
+                    store_update.update_refcount(ColReceiptIdToShardId, &key, &[], -1);
+                    self.chain_store.receipt_id_to_shard_id.cache_remove(&key);
+                    self.inc_gc(ColReceiptIdToShardId);
+                }
+            }
+            Err(error) => {
+                match error.kind() {
+                    ErrorKind::DBNotFoundErr(_) => {
+                        // Sometimes we don't save outgoing receipts. See the usages of save_outgoing_receipt.
```

Comment:
where do we save receipt_id_to_shard_id but not outgoing_receipts?

Reply:
after this change they match

Reply:
I see